netkan/netkan/auto_freezer.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import datetime
+from pathlib import Path
+import git
+
+from .status import ModStatus
+
+
+class AutoFreezer:
+
+    BRANCH_NAME = 'freeze/auto'
+    UNFROZEN_SUFFIX = 'netkan'
+    FROZEN_SUFFIX = 'frozen'
+
+    def __init__(self, netkan_repo, github_pr):
+        self.netkan_repo = netkan_repo
+        self.github_pr = github_pr
+
+    def freeze_idle_mods(self, days_limit):
+        update_cutoff = datetime.datetime.now() - datetime.timedelta(days=days_limit)
+        self.netkan_repo.remotes.origin.pull('master', strategy_option='ours')
+        ids_to_freeze = [ident for ident in self._ids() if self._too_old(ident, update_cutoff)]
+        if ids_to_freeze:
+            self._checkout_branch(self.BRANCH_NAME)
+            for ident in ids_to_freeze:
+                if Path(self.netkan_repo.working_dir,
+                        'NetKAN',
+                        f'{ident}.{self.UNFROZEN_SUFFIX}'
+                        ).exists():
+                    logging.info('Freezing %s', ident)
+                    self._add_freezee(ident)
+                else:
+                    logging.info('Already froze %s', ident)
+            self._submit_pr(self.BRANCH_NAME, days_limit)
+            self.netkan_repo.heads.master.checkout()
+
+    def mark_frozen_mods(self):
+        with ModStatus.batch_write() as batch:
+            logging.info('Marking frozen mods...')
+            for mod in ModStatus.scan(rate_limit=5):
+                if not mod.frozen and self._is_frozen(mod.ModIdentifier):
+                    logging.info('Marking frozen: %s', mod.ModIdentifier)
+                    mod.frozen = True
+                    batch.save(mod)
+            logging.info('Done!')
```

Comment:
This seems like a bit of double handling, but rate limited DynamoDB writes are part of the free tier. We _could_ potentially do this with a webhook, but  I think doing this once a week is ok for now and will catch when we mark things as frozen by hand in the interim.

To note `batch_write` requires some special handling, but we're getting rate limiting on our writes as a consequence of our ModStatus scan.